h2o-core/src/main/java/hex/ModelMetricsRegressionCoxPH.java

```diff
@@ -0,0 +1,231 @@
+              weightedSigma(), modelMetricsRegression.mae() , modelMetricsRegression.rmsle(), modelMetricsRegression.mean_residual_deviance(),
+              _customMetric, stats.c(), stats.nconcordant, stats.discordant(), stats.nties);
+
+
+      if (m!=null) m.addModelMetrics(mm);
+      return mm;
+    }
+
+    private Stats concordance(Model m, Frame fr, Frame adaptFrm, Frame scored) {
+      final Vec startVec = adaptFrm.vec(startVecName);
+      final Vec stopVec = adaptFrm.vec(stopVecName);
+      final Vec statusVec = adaptFrm.lastVec();
+      final Vec estimateVec = scored.lastVec();
+
+      final List<Vec> strataVecs =
+              isStratified ?
+                      Arrays.asList(stratifyBy).stream().map(s -> fr.vec(s) ).collect(toList()) :
+                      Collections.emptyList();
+     
+      return concordance(startVec, stopVec, statusVec, strataVecs, estimateVec);
+    }
+
+    static private boolean isValidComparison(double time1, double time2, boolean event1, boolean event2) {
+      if (time1 == time2) {
+        return event1 != event2;
+      }
+      if (event1 && event2) {
+        return true;
+      }
+      if (event1 && time1 < time2) {
+        return true;
+      }
+      if (event2 &&  time2 < time1) {
+        return true;
+      }
+      return false;
+    }
+
+    static class Stats {
+      final long ntotals;
+      final long nNotNaN;
+      final long nconcordant;
+      final long nties;
+
+      Stats() {
+        this(0, 0, 0, 0);
+      }
+
+      Stats(long ntotals, long nNotNaN, long nconcordant, long nties) {
+        this.ntotals = ntotals;
+        this.nNotNaN = nNotNaN;
+        this.nconcordant = nconcordant;
+        this.nties = nties;
+      }
+
+      double c() {
+        return (nconcordant + 0.5d * nties) / ntotals;
+      }
+      
+      long discordant() {
+        return ntotals - nconcordant - nties;
+      }
+
+      Stats plus(Stats s2) {
+        return new Stats(ntotals + s2.ntotals, nNotNaN + s2.nNotNaN, nconcordant + s2.nconcordant, nties + s2.nties);
+      }
+    }
+    
+    static Stats concordance(final Vec startVec, final Vec stopVec, final Vec eventVec, List<Vec> strataVecs, final Vec estimateVec) {
+      final long length = estimateVec.length();
+
+      final Stats stats = concordanceStats(null == startVec ? null : startVec.new Reader(), 
+              stopVec.new Reader(), eventVec.new Reader(),
+              strataVecs.stream().map(it -> it.new Reader()).collect(toList()),
+              estimateVec.new Reader(), length);
+
+      return stats;
+    }
+
+    private static Stats concordanceStats(Vec.Reader startVec, Vec.Reader stopVec, Vec.Reader eventVec, List<Vec.Reader> strataVecs, Vec.Reader estimateVec, long length) {
+      Stream<Integer> allIndexes = IntStream.range(0, (int) length).boxed();
+
+      Map<List<Long>, List<Integer>> byStrata = allIndexes.collect(groupingBy(
```

Comment:
My understanding this code is not final and performance should be improved - in that case, please link a JIRA for the performance improvement.

Reply:
https://0xdata.atlassian.net/browse/PUBDEV-7736